src/js/consts.js

```diff
@@ -45,17 +60,18 @@ export const visibilityFunctions = {
     const userPermissions = await insights.chrome.getUserPermissions();
     return userPermissions && permissions.every((item) => userPermissions.find(({ permission }) => permission === item));
   },
-  apiRequest: async ({ url, method, accessor, ...options }) =>
-    instance({
+  apiRequest: async ({ url, method, accessor, matcher, ...options }) => {
+    return instance({
       url,
       method: method || 'GET',
       ...options,
     })
-      .then((response) => (accessor ? get(response?.data || response || {}, accessor) : response?.data || response))
+      .then((response) => matchValue(accessor ? get(response || {}, accessor) : response, matcher))
```

Comment:
I'd still keep in here the `response?.data || response`
```JS
.then((response) => matchValue(accessor ? get(response || {}, accessor) : response?.data || response, matcher))
```
This way, we don't have to add `accessor` for each generic API and if the response will be located deeply in the response consumers can use `accessor`

Reply:
@karelhala we actually do. Because of some nested endpoints. What if you want to access nested attribute from payload like this:
```jsx
{
  bar: {
    isEntitled: true
  }
}
```

Catalog and approval for instance have user preferences and metadata on certain API responses. I can imagine using those for this exact purpose. Here an example of how it may look.
```jsx
{
  data: [...]
  meta: {
    user_preferences: {
      canRead: false
    }
  }
}
```